newIDE/app/src/InstancesEditor/SelectionRectangle.js

```diff
@@ -82,28 +155,55 @@ export default class SelectionRectangle {
     this.selectionRectangleEnd = { x, y };
   };
 
-  updateSelectionRectangle = (lastX: number, lastY: number) => {
+  updateSelectionRectangle = (
+    lastX: number,
+    lastY: number
+  ): Array<gdInitialInstance> => {
     if (!this.selectionRectangleStart)
       this.selectionRectangleStart = { x: lastX, y: lastY };
 
     this.selectionRectangleEnd = { x: lastX, y: lastY };
+    this._temporaryInstancesInSelectionRectangle.length = 0;
+    const normalizedSelectionRectangle = getRectangleNormalizedBoundaries({
+      startX: this.selectionRectangleStart.x,
+      startY: this.selectionRectangleStart.y,
+      endX: this.selectionRectangleEnd.x,
+      endY: this.selectionRectangleEnd.y,
+    });
+    this.temporarySelectionRectangleStart = {
+      x: normalizedSelectionRectangle.startX,
+      y: normalizedSelectionRectangle.startY,
+    };
+    this.temporarySelectionRectangleEnd = {
+      x: normalizedSelectionRectangle.endX,
+      y: normalizedSelectionRectangle.endY,
+    };
+
+    this.instances.iterateOverInstances(
+      // $FlowFixMe - gd.castObject is not supporting typings.
+      this.temporarySelector
+    );
+    return this._temporaryInstancesInSelectionRectangle;
   };
 
-  endSelectionRectangle = () => {
+  endSelectionRectangle = (): Array<gdInitialInstance> => {
     if (!this.selectionRectangleStart || !this.selectionRectangleEnd) return [];
 
     this._instancesInSelectionRectangle.length = 0;
-    if (this.selectionRectangleStart.x > this.selectionRectangleEnd.x) {
-      const tmp = this.selectionRectangleStart.x;
-      this.selectionRectangleStart.x = this.selectionRectangleEnd.x;
-      this.selectionRectangleEnd.x = tmp;
-    }
-    if (this.selectionRectangleStart.y > this.selectionRectangleEnd.y) {
-      const tmp = this.selectionRectangleStart.y;
-      this.selectionRectangleStart.y = this.selectionRectangleEnd.y;
-      this.selectionRectangleEnd.y = tmp;
-    }
-
+    const normalizedSelectionRectangle = getRectangleNormalizedBoundaries({
+      startX: this.selectionRectangleStart.x,
+      startY: this.selectionRectangleStart.y,
+      endX: this.selectionRectangleEnd.x,
+      endY: this.selectionRectangleEnd.y,
+    });
+    this.selectionRectangleStart = {
```

Comment:
Ditto, in "semi real time" code like this (or in real time code like in the engine), prefer plain old operations rather than creating objects that will be garbage collected immediately after.

Reply:
Or pass the rectangle to mutate directly to the function.